### ISSUE TITLE
False positive value-keyword-case with native value

### ISSUE THREAD

**Issue by rohm1** [stylelint/stylelint-config-standard#292]

**Comment by ybiquitous** [comment]
Revisit old issues:
- https://github.com/stylelint/stylelint/issues/1089
- https://github.com/stylelint/stylelint/issues/4884

**Comment by ybiquitous** [comment]
Just to confirm, are both valid, e.g., `optimizeLegibility` and `optimizelegibility`?

**Comment by rohm1** [comment]
Not sure, the specs make only mention of the camel case option (https://svgwg.org/svg2-draft/painting.html#TextRenderingProperty). But both Firefox and Chrome seem to accept camel case and lowercase.

**Comment by ybiquitous** [comment]
Umm, if we enable `camelCaseSvgKeywords`, the following valid CSS using the [`currentcolor`](https://www.w3.org/TR/css-color-4/#currentcolor-color) keyword will be reported:

```css
a {
  color: currentcolor;
/*       ^ Expected "currentcolor" to be "currentColor"
 */
}
```

See [this demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcIBUACYAdAdugLpANhAE7wpgCuRRMmeEhRA3FgL5YgA0IAZgJb4wAcgEMAtnEQwAHmIAOAgHRgAzss7gImPgHMEIDJhQp0IIuQHKTpA0aMmAbsPzkYAWgDWMAJ4B3YlFcwYWUYKxQAbRNCHxgiEw40EyDxfABhYJgAZXttAGlvPyIoSwQUHDMYFgBdLCM2TBZ1SC1ebQAxYlFhHD0AK2VNdVhZNUQbYxBlHC8BfF5aMKjumCn42ompmZg52kDNHVcp4UwoYSLFkABmS4UABjuTVhAWIA).

What do you think?

**Comment by rohm1** [comment]
Looks like this is wrong: https://github.com/stylelint/stylelint/blob/main/lib/reference/keywords.js#L265 `currentcolor` is expected to be camel case but the spec you mention states lowercase.

**Comment by ybiquitous** [comment]
Does the SVG spec support `currentColor`, right?

**Comment by rohm1** [comment]
The link you provided is `CSS Color Module Level 4` (https://www.w3.org/TR/css-color-4/#currentcolor-color), keyword is only mentioned in lowercase.
In `CSS Color Module Level 3` (https://www.w3.org/TR/css-color-3/#currentcolor) keyword is written in camel case, and references SVG 1.0 specifications which also use camel case.
Finaly, SVG2 draft also mentions the keyword in camel case: https://svgwg.org/svg2-draft/painting.html#ColorProperty.

To summarize:
- CSS3, SVG1 and SVG2: `currentColor`
- CSS4: `currentcolor`

**Comment by ybiquitous** [comment]
Umm..., it seems we cannot enable `camelCaseSvgKeywords` until fixing the problem. 🤔

**Comment by Mouvedia** [comment]
The _CSS color 4_ tests are all using `currentColor`:

https://github.com/web-platform-tests/wpt/blob/master/css/css-color/currentcolor-001.html
https://github.com/web-platform-tests/wpt/blob/master/css/css-color/currentcolor-002.html
https://github.com/web-platform-tests/wpt/blob/master/css/css-color/currentcolor-003.html

**Comment by nschonni** [comment]
I came across this issue, and trying to trace down any upstream issue I found https://github.com/w3c/csswg-drafts/issues/4599 
which points over that the values are case insensitive https://drafts.csswg.org/css-values-4/#keywords, so fixing them to the camelCase should be safe
